f8a_worker/process.py

```diff
@@ -148,12 +148,26 @@ def add_and_commit_everything(self, message="blank"):
         self.add(self.repo_path)
         self.commit(message=message)
 
-    def archive(self, basename):
+    def archive(self, basename, repo_path=None):
+        """
+        Create an archive; simply calls `git archive`.
+
+        :param basename: str, name of the resulting archive, without file extension (suffix)
+        :param repo_path: str, only add files found under this path to the archive;
+                          default: add all files from the repository (.git/ is always excluded)
```

Comment:
I'd suggest using another name (like just `path`), because there already is `self.repo_path`.
`path` is subdirectory (or file) in `self.repo_path`.